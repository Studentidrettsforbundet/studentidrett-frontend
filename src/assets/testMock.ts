// ------------------- REGIONS ---------------
import {CITY, CLUB} from '../constants';

export const singleRegion = { id: 3, name: 'Vestlandet', cities: [] };
export const regionList3 = [
    { id: 0, name: 'Midtnorge', cities: [] },
    { id: 1, name: 'Oslo/Viken', cities: [] },
    { id: 2, name: 'Nord norge', cities: [] },
];
export const regionList5 = [
    { id: 0, name: 'Midtnorge', cities: [] },
    { id: 1, name: 'Oslo/Viken', cities: [] },
    { id: 2, name: 'Nord norge', cities: [] },
    { id: 3, name: 'Vestlandet', cities: [] },
    { id: 4, name: 'Sørlandet', cities: [] },
];

// ------------------- CITIES ----------------
export const singleCity = { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] };
export const cityList = [
    { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] },
    { id: 1, name: 'Oslo', region: 'Oslo/Viken', clubs: [] },
    { id: 2, name: 'Bergen', region: 'Vestlandet', clubs: [] },
    { id: 3, name: 'Stavanger', region: 'Vestlandet', clubs: [] },
];

export const invalidCityList = [
    { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] },
    { id: 1, name: 'Oslo', clubs: [] },
];
export const newCity = { id: 5, name: 'Orkanger', region: 'Midtnorge', clubs: [] };

// ------------------- CLUBS -----------------
export const singleClub = {
    id: 3,
    city: 1,
    name: 'NTNUI Bowling',
    description: '',
    contact_email: 'info@ntnui.no',
    membership_fee: '200',
    register_info: 'Register at ntnui.no/register',
};
export const clubList2 = [
    {
        id: 0,
        city: 0,
        name: 'NTNUI football',
        description: '',
        contact_email: 'info@ntnui.no',
        membership_fee: '200',
        register_info: 'Register at ntnui.no/register',
    },
    {
        id: 1,
        city: 0,
        name: 'NTNUI amerikansk fotball',
        description: '',
        contact_email: 'info@ntnui.no',
        membership_fee: '200',
        register_info: 'Register at ntnui.no/register',
    },
];
export const newClub = {
    id: 0,
    city: 0,
    name: 'NTNUI football',
    description: '',
    contact_email: 'info@ntnui.no',
    membership_fee: '200',
    register_info: 'Register at ntnui.no/register',
};

// ------------------- GROUPS ----------------
export const singleGroup = {
    id: 33,
    name: 'ntnui hockey',
    description: 'A hockey group at NTNUI',
    cover_photo: 'photo',
    club: 0,
    city: 3,
    contact_email: 'hockey@ntnui.no',
    sports: [],
};

export const groupList1 = [
    {
        id: 0,
        name: 'ntnui football',
        description: 'A football group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 0,
        contact_email: 'football@ntnui.no',
        sports: [],
    },
];

export const groupList2 = [
    {
        id: 33,
        name: 'ntnui hockey',
        description: 'A hockey group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 3,
        contact_email: 'hockey@ntnui.no',
        sports: [],
    },
    {
        id: 0,
        name: 'ntnui football',
        description: 'A football group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 0,
        contact_email: 'football@ntnui.no',
        sports: [],
    },
];

export const newGroup = {
    id: 0,
    name: 'ntnui football',
    description: 'A football group at NTNUI',
    cover_photo: 'photo',
    club: 0,
    city: 0,
    contact_email: 'football@ntnui.no',
    sports: [],
};

// ------------------- TEAMS -----------------
export const singleTeam = {
    id: 0,
    availability: 'OP',
    cost: null,
    equipment: null,
    facebook_link: null,
    gender: 'M',
    group: 1,
    image: null,
    instagram_link: null,
    location: 1,
    long_description: 'Team A is the best',
    name: 'Team A',
    schedule: [],
    tryout_dates: [],
    webpage: null,
    season: null,
    short_description: 'Short desc',
    skill_level: 'HI',
    sport: 1,
};
export const newTeam = {
    id: 4,
    availability: 'OP',
    cost: null,
    equipment: null,
    facebook_link: null,
    gender: 'M',
    group: 1,
    image: null,
    instagram_link: null,
    location: 1,
    long_description: 'Team A is the best',
    name: 'Team D',
    schedule: [],
    tryout_dates: [],
    webpage: null,
    season: null,
    short_description: 'Short desc',
    skill_level: 'HI',
    sport: 1,
};
export const teamList2 = [
    {
        id: 0,
        availability: 'OP',
        cost: null,
        equipment: null,
        facebook_link: null,
        gender: 'M',
        group: 1,
        image: null,
        instagram_link: null,
        location: 1,
        long_description: 'Team A is the best',
        name: 'Team A',
        schedule: [],
        tryout_dates: [],
        webpage: null,
        season: null,
        short_description: 'Short desc',
        skill_level: 'HI',
        sport: 1,
    },
    {
        id: 2,
        availability: 'OP',
        cost: null,
        equipment: null,
        facebook_link: null,
        gender: 'M',
        group: 1,
        image: null,
        instagram_link: null,
        location: 1,
        long_description: 'Team A is the best',
        name: 'Team B',
        schedule: [],
        tryout_dates: [],
        webpage: null,
        season: null,
        short_description: 'Short desc',
        skill_level: 'HI',
        sport: 1,
    },
];

// ------------------- SPORTS ----------------
export const singleSport = { id: 0, name: 'Football', labels: [] };
export const newSport = { id: 6, name: 'Tennis', labels: [] };
export const sportList2 = [
    { id: 0, name: 'Football', labels: [] },
    { id: 3, name: 'American football', labels: [] },
];
export const sportList5 = [
    {
        id: 0,
        name: 'Football',
        labels: [
            {
                text: 'utendørs',
                sports: ['Fotball', 'Langrenn', 'Rumpeldunk'],
                alternatives: ['Utendørs'],
            },
        ],
    },

    {
        id: 1,
        name: 'Handball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Hanball'],
                alternatives: ['Indoors'],
            },
        ],
    },
    {
        id: 2,
        name: 'Basketball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Basketball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
    {
        id: 3,
        name: 'American football',
        labels: [
            {
                text: 'utendørs',
                sports: ['Fotball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
    {
        id: 4,
        name: 'Baseball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Baseball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
];

// ------------------- SEARCH ----------------
export const searchResult = { name: 'Test', id: '1', label: CITY };
export const newSearchResult = { name: 'Test2', id: '2', label: CLUB };
