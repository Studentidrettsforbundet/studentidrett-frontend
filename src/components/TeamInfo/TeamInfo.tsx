import React, { useState } from 'react';
import './TeamInfo.css';

const ClubInfo = (props: { name: string; img: string; text: string, showInfo: Function }) => {
    
    var [showInfo, setShowInfo] = useState(false);

    if (showInfo == true) {
        return (
            <div className="info">
                <h1>{props.name}</h1>
                <img className="teamImg" src={require(`../../assets/${props.img}.png`)} alt="team img" />
                <p>{props.text}</p>
                <button onClick={() => setShowInfo(false)}> show less </button>
            </div>);
        }
    return (
        <div className="info">
            <h1>{props.name}</h1>
            <img className="teamImg" src={require(`../../assets/${props.img}.png`)} alt="team img" />
            <button onClick={() => setShowInfo(true)}> show more </button>
        </div>
    );
};
export default ClubInfo;
