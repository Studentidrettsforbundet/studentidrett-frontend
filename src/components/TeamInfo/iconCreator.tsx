import React from 'react';

export const getLinkIcon = (site: string | null): JSX.Element | null => {
    if (site === 'Facebook') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
                <title>Logo Facebook</title>
                <path
                    d="M480 257.35c0-123.7-100.3-224-224-224s-224 100.3-224 224c0 111.8 81.9 204.47 189 221.29V322.12h-56.89v-64.77H221V208c0-56.13 33.45-87.16 84.61-87.16 24.51 0 50.15 4.38 50.15 4.38v55.13H327.5c-27.81 0-36.51 17.26-36.51 35v42h62.12l-9.92 64.77H291v156.54c107.1-16.81 189-109.48 189-221.31z"
                    fillRule="evenodd"
                    stroke="currentColor"
                    fill="currentColor"
                />
            </svg>
        );
    } else if (site === 'Instagram') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
                <title>Logo Instagram</title>
                <path
                    d="M349.33 69.33a93.62 93.62 0 0193.34 93.34v186.66a93.62 93.62 0 01-93.34 93.34H162.67a93.62 93.62 0 01-93.34-93.34V162.67a93.62 93.62 0 0193.34-93.34h186.66m0-37.33H162.67C90.8 32 32 90.8 32 162.67v186.66C32 421.2 90.8 480 162.67 480h186.66C421.2 480 480 421.2 480 349.33V162.67C480 90.8 421.2 32 349.33 32z"
                    stroke="currentColor"
                    fill="currentColor"
                />
                <path
                    d="M377.33 162.67a28 28 0 1128-28 27.94 27.94 0 01-28 28zM256 181.33A74.67 74.67 0 11181.33 256 74.75 74.75 0 01256 181.33m0-37.33a112 112 0 10112 112 112 112 0 00-112-112z"
                    stroke="currentColor"
                    fill="currentColor"
                />
            </svg>
        );
    } else if (site === 'Hjemmeside') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
                <title>Globe</title>
                <path
                    d="M256 48C141.13 48 48 141.13 48 256s93.13 208 208 208 208-93.13 208-208S370.87 48 256 48z"
                    fill="none"
                    stroke="currentColor"
                    strokeMiterlimit="10"
                    strokeWidth="32"
                />
                <path
                    d="M256 48c-58.07 0-112.67 93.13-112.67 208S197.93 464 256 464s112.67-93.13 112.67-208S314.07 48 256 48z"
                    fill="none"
                    stroke="currentColor"
                    strokeMiterlimit="10"
                    strokeWidth="32"
                />
                <path
                    d="M117.33 117.33c38.24 27.15 86.38 43.34 138.67 43.34s100.43-16.19 138.67-43.34M394.67 394.67c-38.24-27.15-86.38-43.34-138.67-43.34s-100.43 16.19-138.67 43.34"
                    fill="none"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="32"
                />
                <path
                    fill="none"
                    stroke="currentColor"
                    strokeMiterlimit="10"
                    strokeWidth="32"
                    d="M256 48v416M464 256H48"
                />
            </svg>
        );
    } else {
        return null;
    }
};
export const getGenderIcon = (gender: string | null): JSX.Element | null => {
    if (gender === 'M') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" stroke="currentColor" fill="currentColor">
                <title>Man</title>
                <circle cx="256" cy="56" r="56" />
                <path d="M304 128h-96a64.19 64.19 0 00-64 64v107.52c0 10.85 8.43 20.08 19.27 20.47A20 20 0 00184 300v-99.73a8.18 8.18 0 017.47-8.25 8 8 0 018.53 8V489a23 23 0 0023 23 23 23 0 0023-23V346.34a10.24 10.24 0 019.33-10.34A10 10 0 01266 346v143a23 23 0 0023 23 23 23 0 0023-23V200.27a8.18 8.18 0 017.47-8.25 8 8 0 018.53 8v99.52c0 10.85 8.43 20.08 19.27 20.47A20 20 0 00368 300V192a64.19 64.19 0 00-64-64z" />
            </svg>
        );
    } else if (gender === 'F') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" stroke="currentColor" fill="currentColor">
                <title>Woman</title>
                <circle cx="255.75" cy="56" r="56" />
                <path d="M394.63 277.9l-10.33-34.41v-.11l-22.46-74.86h-.05l-2.51-8.45a44.87 44.87 0 00-43-32.08h-120a44.84 44.84 0 00-43 32.08l-2.51 8.45h-.06l-22.46 74.86v.11l-10.37 34.41c-3.12 10.39 2.3 21.66 12.57 25.14a20 20 0 0025.6-13.18l25.58-85.25 2.17-7.23a8 8 0 0115.53 2.62 7.78 7.78 0 01-.17 1.61L155.43 347.4a16 16 0 0015.32 20.6h29v114.69c0 16.46 10.53 29.31 24 29.31s24-12.85 24-29.31V368h16v114.69c0 16.46 10.53 29.31 24 29.31s24-12.85 24-29.31V368h30a16 16 0 0015.33-20.6l-43.74-145.81a7.52 7.52 0 01-.16-1.59 8 8 0 0115.54-2.63l2.17 7.23 25.57 85.25A20 20 0 00382.05 303c10.27-3.44 15.69-14.71 12.58-25.1z" />
            </svg>
        );
    } else if (gender === 'MX' || gender === 'A') {
        return (
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512" stroke="currentColor" fill="currentColor">
                <title>Male Female</title>
                <path d="M426 16h-74a22 22 0 000 44h20.89l-37.1 37.09A157.68 157.68 0 00216 42c-87.12 0-158 70.88-158 158 0 79.66 59.26 145.72 136 156.46V394h-28a22 22 0 000 44h28v36a22 22 0 0044 0v-36h28a22 22 0 000-44h-28v-37.54c76.74-10.74 136-76.8 136-156.46a157.15 157.15 0 00-14-64.92l44-44V112a22 22 0 0044 0V38a22 22 0 00-22-22zM216 314a114 114 0 11114-114 114.13 114.13 0 01-114 114z" />
            </svg>
        );
    } else {
        return null;
    }
};

export const getSkillIcon = (skill: string | null): JSX.Element | null => {
    if (skill === 'NONE') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-reception-0"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M0 13.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5z" />
            </svg>
        );
    } else if (skill === 'LOW') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-reception-1"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M0 11.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-2zm4 2a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5z" />
            </svg>
        );
    } else if (skill === 'MED') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-reception-2"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M0 11.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-2zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-5zm4 5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5zm4 0a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5z" />
            </svg>
        );
    } else if (skill === 'HI') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-reception-3"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M0 11.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-2zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-5zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-8zm4 8a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 0 1h-2a.5.5 0 0 1-.5-.5z" />
            </svg>
        );
    } else if (skill === 'PRO') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-reception-4"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path d="M0 11.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-2zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-5zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-8zm4-3a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5v11a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1-.5-.5v-11z" />
            </svg>
        );
    } else {
        return null;
    }
};

export const getAvailabilityIcon = (availability: string | null): JSX.Element | null => {
    if (availability === 'OP' || availability === 'TO') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-person-plus-fill"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    fillRule="evenodd"
                    d="M1 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H1zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm7.5-3a.5.5 0 0 1 .5.5V7h1.5a.5.5 0 0 1 0 1H14v1.5a.5.5 0 0 1-1 0V8h-1.5a.5.5 0 0 1 0-1H13V5.5a.5.5 0 0 1 .5-.5z"
                />
            </svg>
        );
    } else if (availability === 'CL' || availability === 'FU') {
        return (
            <svg
                width="1em"
                height="1em"
                viewBox="0 0 16 16"
                className="bi bi-person-x-fill"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
            >
                <path
                    fillRule="evenodd"
                    d="M1 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H1zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm6.146-2.854a.5.5 0 0 1 .708 0L14 6.293l1.146-1.147a.5.5 0 0 1 .708.708L14.707 7l1.147 1.146a.5.5 0 0 1-.708.708L14 7.707l-1.146 1.147a.5.5 0 0 1-.708-.708L13.293 7l-1.147-1.146a.5.5 0 0 1 0-.708z"
                />
            </svg>
        );
    } else {
        return null;
    }
};
