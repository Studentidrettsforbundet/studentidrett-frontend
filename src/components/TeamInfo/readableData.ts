export const getTeamGender = (gender: string | null) => {
    const genders: { [index: string]: string } = {
        M: 'Herre',
        F: 'Dame',
        MX: 'Mix',
        A: 'Alle',
    };
    if (gender) {
        return genders[gender];
    }
    return null;
};

export const getTeamSkill = (skill: string | null) => {
    const skills: { [index: string]: string } = {
        NONE: 'Ingen forkunnskaper krevd',
        LOW: 'Lavt',
        MED: 'Medium',
        HI: 'Høyt',
        Pro: 'Proff',
    };
    if (skill) {
        return skills[skill];
    }
    return null;
};

export const getTeamAvailability = (availability: string | null) => {
    const translations: { [index: string]: string } = {
        OP: 'Åpent for alle',
        CL: 'Opptak ferdig',
        FU: 'Laget er fullt',
        TO: 'Vi har opptak',
    };
    if (availability) {
        return translations[availability];
    }
    return null;
};
