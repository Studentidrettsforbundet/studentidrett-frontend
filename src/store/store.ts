import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
