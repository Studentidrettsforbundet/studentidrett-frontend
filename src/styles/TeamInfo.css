.info {
    margin: 16px 48px;
    padding: 16px;
    background-color: aliceblue;
    width: 100%;
}

.teamImg {
    max-width: 100%;
    height: auto;
}
