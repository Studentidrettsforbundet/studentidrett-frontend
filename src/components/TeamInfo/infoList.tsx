import React from 'react';
import { getTeamGender, getTeamSkill, getTeamAvailability } from './readableData';
import { getAvailabilityIcon, getGenderIcon, getSkillIcon } from './iconCreator';
import { icon, list, listItem } from './styles';

interface InfoListProps {
    availability: string | null;
    gender: string | null;
    skill_level: string | null;
}

export const InfoList = ({ availability, gender, skill_level }: InfoListProps): JSX.Element | null => {
    const items = [
        { item: 'Status', text: getTeamAvailability(availability), icon: getAvailabilityIcon(availability) },
        { item: 'Kjønn', text: getTeamGender(gender), icon: getGenderIcon(gender) },
        { item: 'Nivå', text: getTeamSkill(skill_level), icon: getSkillIcon(skill_level) },
    ];

    const renderedList = items.map((obj, i) => {
        if (obj.text) {
            return (
                <div className={listItem} key={i}>
                    <b>{obj.item}:</b> {obj.text}
                    <div className={icon}>{obj.icon}</div>
                </div>
            );
        }
    });
    return renderedList ? (
        <div className={list}>
            {renderedList}
            <hr />
        </div>
    ) : null;
};
