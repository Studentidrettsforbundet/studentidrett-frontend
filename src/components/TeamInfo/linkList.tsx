import React from 'react';
import { getLinkIcon } from './iconCreator';
import { icon, list, linkItem } from './styles';

interface LinkListProps {
    facebook_link: string | null;
    instagram_link: string | null;
    webpage: string | null;
}

export const LinkList = ({ facebook_link, instagram_link, webpage }: LinkListProps): JSX.Element | null => {
    const links = [
        { site: facebook_link, text: 'Facebook' },
        { site: instagram_link, text: 'Instagram' },
        { site: webpage, text: 'Hjemmeside' },
    ];

    const renderedLinks = links.map((link, i) => {
        if (link.site) {
            return (
                <a className={linkItem} key={i} href={link.site}>
                    <div className={icon}>{getLinkIcon(link.text)}</div>
                    {link.text}
                </a>
            );
        }
        return <React.Fragment key={i} />;
    });

    return renderedLinks ? <div className={list}>{renderedLinks}</div> : null;
};
