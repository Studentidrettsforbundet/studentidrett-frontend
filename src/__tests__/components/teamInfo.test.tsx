import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import TeamInfo from '../../components/TeamInfo/teamInfo';
import store from '../../store/store';
import { singleTeam } from '../../assets/testMock';

describe('clubCard', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/sports/1']}>
                    <TeamInfo {...singleTeam} />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
