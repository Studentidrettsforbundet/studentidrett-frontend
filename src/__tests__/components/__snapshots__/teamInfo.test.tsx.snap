// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clubCard renders correctly 1`] = `
Array [
  <div>
    <img
      alt="Team A"
      src="placeholder.png"
    />
  </div>,
  <div>
    <h1>
      Team A
    </h1>
  </div>,
  <div>
    <p>
      Team A is the best
    </p>
  </div>,
]
`;
